src/Mutex.php

```diff
@@ -44,55 +32,42 @@ public function __construct(bool $autoRelease = true)
         if ($autoRelease) {
             $locks = &$this->locks;
             register_shutdown_function(function () use (&$locks) {
-                /**
-                 * @var string $lock
-                 */
+                /** @var string $lock */
                 foreach ($locks as $lock) {
                     $this->release($lock);
                 }
             });
         }
     }
 
-    /**
-     * Acquires a lock by name.
-     *
-     * @param string $name Name of the lock to be acquired. Must be unique.
-     * @param int $timeout Time (in seconds) to wait for lock to be released. Defaults to zero meaning that method
-     * will return false immediately in case lock was already acquired.
-     *
-     * @return bool lock acquiring result.
-     */
-    public function acquire(string $name, int $timeout = 0): bool
+    public function acquire(string $name, int $timeout = 0): MutexLockInterface
     {
         if (!in_array($name, $this->locks, true) && $this->acquireLock($name, $timeout)) {
             $this->locks[] = $name;
 
-            return true;
+            return new MutexLock($this, $name);
         }
 
-        return false;
+        throw new MutexLockedException();
     }
 
     /**
      * Releases acquired lock. This method will return false in case the lock was not found.
      *
      * @param string $name Name of the lock to be released. This lock must already exist.
-     *
-     * @return bool Lock release result: false in case named lock was not found.
      */
-    public function release(string $name): bool
+    public function release(string $name): void
```

Comment:
Either it should be part of the interface or should be removed.

Reply:
The question here is "do we need global locks that could be accessed by name from everywhere or scope is usually small enough to pass the lock object around?".

Reply:
Reviewed again. Got the idea.